### PR TITLE
Cleanup dev env network interfaces

### PR DESCRIPTION
**Pull request by Sunnatillo**
This PR adds scripts to remove network interfaces added by dev env in host cleanup step.
Fixes: #1436

### CODE REVIEW

File: host_cleanup.sh [hunk 1]
```diff
@@ -51,13 +51,24 @@ ANSIBLE_FORCE_COLOR=true "${ANSIBLE}-playbook" \
...
+    sudo nmcli con delete ironicendpoint
     sudo nmcli con delete provisioning
   fi
   # External net should have been cleaned already at this stage, but we double
   # check as leaving it around causes issues when the host is rebooted
   if [ "${MANAGE_EXT_BRIDGE}" == "y" ]; then
     sudo nmcli con delete external || true
   fi
+else
+  if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
```
Comment (tuminoid) [inline host_cleanup.sh:64]:
```suggestion
  if [[ "${MANAGE_PRO_BRIDGE}" == "y" ]]; then
```

File: host_cleanup.sh [hunk 2]
```diff
@@ -51,13 +51,24 @@ ANSIBLE_FORCE_COLOR=true "${ANSIBLE}-playbook" \
...
+    sudo ip link delete ironic-peer 
+    sudo ip link delete ironicendpoint
+    sudo ip link delete provisioning
+  fi
+    if [ "${MANAGE_EXT_BRIDGE}" == "y" ]; then
```
Comment (tuminoid) [inline host_cleanup.sh:69]:
```suggestion
  if [[ "${MANAGE_EXT_BRIDGE}" == "y" ]]; then
```

File: host_cleanup.sh [hunk 3]
```diff
@@ -51,13 +51,24 @@ ANSIBLE_FORCE_COLOR=true "${ANSIBLE}-playbook" \
 if [[ $OS == "centos" || $OS == "rhel" ]]; then
   sudo rm -rf /etc/NetworkManager/conf.d/dnsmasq.conf
   if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
+    sudo nmcli con delete ironic-peer 
+    sudo nmcli con delete ironicendpoint
```
Comment (tuminoid) [inline host_cleanup.sh:55]: Ironicendpoint is not hardcoded, but is `BARE_METAL_PROVISIONER_INTERFACE`

File: host_cleanup.sh [hunk 4]
```diff
@@ -51,13 +51,24 @@ ANSIBLE_FORCE_COLOR=true "${ANSIBLE}-playbook" \
...
   fi
   # External net should have been cleaned already at this stage, but we double
   # check as leaving it around causes issues when the host is rebooted
   if [ "${MANAGE_EXT_BRIDGE}" == "y" ]; then
     sudo nmcli con delete external || true
   fi
+else
+  if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
+    sudo ip link delete ironic-peer 
+    sudo ip link delete ironicendpoint
```
Comment (tuminoid) [inline host_cleanup.sh:66]: same

**Review by tuminoid — COMMENTED** [review]
Also, add description and proper commit message in Git.

**Comment by mboukhalfa** [comment]
> _ No description provided. _

Please add a discription

**Comment by Sunnatillo** [comment]
/hold 
Project infra Pr should merge first: 
https://github.com/metal3-io/project-infra/pull/794

File: host_cleanup.sh [hunk 5]
```diff
@@ -50,14 +50,25 @@ ANSIBLE_FORCE_COLOR=true "${ANSIBLE}-playbook" \
 # There was a bug in this file, it may need to be recreated.
 if [[ $OS == "centos" || $OS == "rhel" ]]; then
   sudo rm -rf /etc/NetworkManager/conf.d/dnsmasq.conf
-  if [ "$MANAGE_PRO_BRIDGE" == "y" ]; then
+  if [[  "$MANAGE_PRO_BRIDGE" == "y" ]]; then
```
Comment (tuminoid) [inline host_cleanup.sh:53]:
```suggestion
  if [[  "${MANAGE_PRO_BRIDGE}" == "y" ]]; then
```

File: host_cleanup.sh [hunk 6]
```diff
@@ -50,14 +50,25 @@ ANSIBLE_FORCE_COLOR=true "${ANSIBLE}-playbook" \
...
     sudo nmcli con delete provisioning
   fi
   # External net should have been cleaned already at this stage, but we double
   # check as leaving it around causes issues when the host is rebooted
-  if [ "${MANAGE_EXT_BRIDGE}" == "y" ]; then
+  if [[ "${MANAGE_EXT_BRIDGE}" == "y" ]]; then
     sudo nmcli con delete external || true
   fi
+else
+  if [[ "$MANAGE_PRO_BRIDGE" == "y" ]]; then
```
Comment (tuminoid) [inline host_cleanup.sh:64]:
```suggestion
  if [[ "${MANAGE_PRO_BRIDGE}" == "y" ]]; then
```

File: lib/common.sh [hunk 7]
```diff
@@ -293,6 +293,8 @@ else
   export CAPM3_IMAGE="${CAPM3_IMAGE:-${CONTAINER_REGISTRY}/metal3-io/cluster-api-provider-metal3:main}"
   export IPAM_IMAGE="${IPAM_IMAGE:-${CONTAINER_REGISTRY}/metal3-io/ip-address-manager:main}"
   export BMOBRANCH="${BMORELEASEBRANCH:-main}"
+  export KEEPALIVED_TAG="v0.6.1"
```
Comment (tuminoid) [inline lib/common.sh:296]: These are added for testing, or actually part of the change?

File: host_cleanup.sh (hunk 3 above)
Comment (tuminoid) [reply host_cleanup.sh]: Fixed

File: host_cleanup.sh (hunk 4 above)
Comment (tuminoid) [reply host_cleanup.sh]: Fixed

File: lib/common.sh (hunk 7 above)
Comment (Sunnatillo) [reply lib/common.sh]: These were for debugging. I removed them.

File: host_cleanup.sh (hunk 6 above)
Comment (Sunnatillo) [reply host_cleanup.sh]: done

File: host_cleanup.sh (hunk 5 above)
Comment (Sunnatillo) [reply host_cleanup.sh]: done

**Comment by Sunnatillo** [comment]
/test metal3-bml-integration-test-centos

ssh issue

**Review by tuminoid — COMMENTED** [review]
/lgtm

**Review by Rozzii — COMMENTED** [review]
/approve

**Pull request merged**